integration/src/test/resources/docker/docker-compose.yaml

```diff
@@ -49,3 +48,16 @@ services:
        - -force
        - -a
        - "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=${SOLR_DEBUG_PORT}"
+
+  rdf4j:
+    container_name: rdf4j
+    image: jetty:9.4.7-alpine
+    user: root
+    env_file: .env
+    environment:
+      JAVA_OPTIONS: "-Dorg.eclipse.rdf4j.appdata.basedir=/tmp/rdf4j/data -Djetty.http.port=${RDF4J_PORT}"
+    ports:
+      - "${RDF4J_PORT}:${RDF4J_PORT}"
+    volumes:
+      - "./rdf4j/data:/tmp/rdf4j/data"
```

Comment:
This is giving me problems when doing `mvn clean install` - my data folder is here: 
`- "../data:/tmp/rdf4j/data"`
If I change it to this, integration tests are successful. Here is the error:
```
[ERROR] Failed to execute goal io.fabric8:docker-maven-plugin:0.23.0:start (start-docker-its) on project rmap-integration: I/O Error: Unable to create container for [jetty:9.4.7-alpine] : Mount denied:
[ERROR] The source path "C:/Users/uname/Documents/rmap-rdf4j-docker-pr-prep/integration/target/test-classes/docker/rdf4j/data"
```

Reply:
Hm.  The paths should be the same.  

Does the directory `C:/Users/uname` actually exist on your platform?

Reply:
The `data` directory should be present under `integration/target/test-classes/docker/rdf4j` after Maven's `process-test-resources` phase runs.  

After running `mvn clean` (in `integration/`), then run `mvn process-test-resources` and look at the directory structure under `target/test-classes`:

```
esm:~/workspaces/rmap/rmap/integration (pr229)$ mvn process-test-resources
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Building RMap Integration 2.0.1-beta-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- build-helper-maven-plugin:3.0.0:reserve-network-port (reserve-port) @ rmap-integration ---
[INFO] Reserved port 65106 for rmap.webapp.test.port
[INFO] Reserved port 65107 for db.port
[INFO] Reserved port 65108 for rdf4j.port
[INFO] 
[INFO] --- maven-remote-resources-plugin:1.5:process (process-remote-resources) @ rmap-integration ---
[INFO] 
[INFO] --- maven-resources-plugin:3.0.2:resources (default-resources) @ rmap-integration ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 8 resources
[INFO] Copying 78 resources
[INFO] 
[INFO] --- maven-compiler-plugin:2.5.1:compile (default-compile) @ rmap-integration ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-dependency-plugin:3.0.2:copy (copy-rdf4j-wars) @ rmap-integration ---
[INFO] Configured Artifact: org.eclipse.rdf4j:rdf4j-http-server:2.2.2:war
[INFO] Configured Artifact: org.eclipse.rdf4j:rdf4j-http-workbench:2.2.2:war
[INFO] Copying rdf4j-http-server-2.2.2.war to /Users/esm/workspaces/rmap/rmap/integration/target/test-classes/docker/rdf4j/webapps/rdf4j-server.war
[INFO] Copying rdf4j-http-workbench-2.2.2.war to /Users/esm/workspaces/rmap/rmap/integration/target/test-classes/docker/rdf4j/webapps/rdf4j-workbench.war
[INFO] 
[INFO] --- maven-resources-plugin:3.0.2:testResources (default-testResources) @ rmap-integration ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 8 resources
[INFO] Copying 88 resources
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 5.939 s
[INFO] Finished at: 2018-11-30T16:48:40-05:00
[INFO] Final Memory: 67M/2213M
[INFO] ------------------------------------------------------------------------
esm:~/workspaces/rmap/rmap/integration (pr229)$ tree target/test-classes/
target/test-classes/
├── agents
│   ├── agentA.jsonld
│   ├── agentA.rdf
│   ├── agentA.ttl
│   ├── agentA_noauthid.rdf
│   ├── agentA_noid.rdf
│   ├── agentA_noidprovider.rdf
│   ├── agentA_noname.rdf
│   ├── agentA_notype.rdf
│   ├── agentB_v1.rdf
│   ├── agentB_v2.rdf
│   └── agentB_v3.rdf
├── cores
│   └── discos
│       ├── conf
│       │   ├── disco-schema.xml
│       │   ├── elevate.xml
│       │   └── solrconfig.xml
│       └── core.properties
├── create-rmap-agent.sql
├── data
│   └── discos
│       ├── 931zcrjgzb
│       │   ├── 931zcrjgzb.n4
│       │   ├── README.md
│       │   ├── rmp1835x5x.n4
│       │   └── rmp2543q5.n4
│       ├── jwstqjq4mq
│       │   ├── 9cnp5hqdb7.n4
│       │   ├── README.md
│       │   ├── gtht76hgmh.n4
│       │   ├── jwstqjq4mq.n4
│       │   └── rmd18mddgf.n4
│       ├── rmd18mddcw
│       │   ├── README.md
│       │   ├── rmd18m7mj4.n4
│       │   ├── rmd18m7mj4.ttl
│       │   ├── rmd18m7mr7.n4
│       │   ├── rmd18m7mr7.ttl
│       │   ├── rmd18m7msr.n4
│       │   ├── rmd18m7msr.ttl
│       │   ├── rmd18mdd8b.n4
│       │   ├── rmd18mdd8b.ttl
│       │   ├── rmd18mdd9v.n4
│       │   ├── rmd18mdd9v.ttl
│       │   ├── rmd18mddcw.n4
│       │   ├── rmd18mddcw.ttl
│       │   ├── rmd18mdddd.n4
│       │   └── rmd18mdddd.ttl
│       ├── rmd18mddcw-with-tombstone
│       │   ├── README.md
│       │   ├── rmd18m7mj4.n4
│       │   ├── rmd18m7mr7.n4
│       │   ├── rmd18m7msr.n4
│       │   ├── rmd18mdd8b.n4
│       │   ├── rmd18mdd9v.n4
│       │   ├── rmd18mddcw.n4
│       │   ├── rmd18mdddd.n4
│       │   └── rmp1892fdx.n4
│       ├── rmp1892f7b
│       │   ├── rmp1892f6t.ttl
│       │   └── rmp1892f7b.ttl
│       └── rmp1892gbc
│           ├── README.md
│           ├── rmd18m7mj4.n4
│           ├── rmp1892fdx.n4
│           ├── rmp1892g9v.n4
│           └── rmp1892gbc.n4
├── discos
│   ├── discoA.jsonld
│   ├── discoA.rdf
│   ├── discoA.ttl
│   ├── discoA_aggregatesonly.rdf
│   ├── discoA_badsyntax.rdf
│   ├── discoA_encodedspaceinurl.rdf
│   ├── discoA_noaggregates.rdf
│   ├── discoA_nobody.rdf
│   ├── discoA_nobody_noaggregates.rdf
│   ├── discoA_nonaggregateroots.ttl
│   ├── discoA_notconnected.rdf
│   ├── discoA_notype.rdf
│   ├── discoA_withbnodes.jsonld
│   ├── discoA_withbnodes.rdf
│   ├── discoA_withbnodes.ttl
│   ├── discoA_withproviderid.rdf
│   ├── discoA_x2.rdf
│   ├── discoB_v1.rdf
│   ├── discoB_v2.rdf
│   ├── discoB_v3.rdf
│   └── discoB_v4.rdf
├── docker
│   ├── cores
│   │   ├── discos
│   │   │   └── data
│   │   │       └── index
│   │   │           ├── segments_1
│   │   │           └── write.lock
│   │   └── versions
│   │       └── data
│   ├── docker-compose.yaml
│   └── rdf4j
│       ├── jetty.start
│       └── webapps
│           ├── rdf4j-server.war
│           └── rdf4j-workbench.war
├── empty.txt
└── logging.properties

22 directories, 85 files
esm:~/workspaces/rmap/rmap/integration (pr229)$ 
```

Reply:
yes, I anonymized the path. The rdf4j folder exists, but the data folder is in the same folder as docker: 

![image](https://user-images.githubusercontent.com/4070836/49316675-a5a84780-f4bf-11e8-9e45-a4586614038e.png)

![image](https://user-images.githubusercontent.com/4070836/49316704-bd7fcb80-f4bf-11e8-9de4-1e3d6ccf9a52.png)

![image](https://user-images.githubusercontent.com/4070836/49316728-d9836d00-f4bf-11e8-9bd2-45ad30fd2ed9.png)

Reply:
Oh maybe there is an issue... let's see.

Reply:
Your directory structure looks the same as mine. I have no idea why it passes travis but not locally - weird

Reply:
@karenhanson there are a couple of `data` directories.  The `target/test-classes/docker/rdf4j/data/` is created when `docker:start` runs.  I'm guessing your platform isn't automatically creating the directory.

What happens if you upgrade the `docker-maven-plugin` to version to `0.27.2`?

Reply:
to clarify my original comment, if I change the current volume mount path from this:
`- "./rdf4j/data:/tmp/rdf4j/data"`
to this:
`- "../data:/tmp/rdf4j/data"`
everything works great

Reply:
@karenhanson `target/test-classes/data` is a different directory, and isn't the one being referenced in the `docker-compose.yaml`.  The paths present in `docker-compose.yaml` are relative to `integration/target/test-classes/docker`.

Reply:
oh ok! is the data directory supposed to be empty then? Maybe with Windows it refuses to create the directory if isn't there already?

Reply:
@karenhanson yeah, I believe it is supposed to be empty.  It may be Windows, or it may be the `docker-maven-plugin`... so that's why I suggest attempting to upgrade the plugin and see if that fixes it.

Reply:
@emetsger Sorry the conversations crossed over and I didn't see that you had already come to the same conclusion of missing directory! 

So, adding `rdf4j/data/.gitkeep` worked. Using latest `docker-maven-plugin` did not unfortunately. So I guess we found another Windows specific quirk - good to know! Once that file is added I think this is good to merge.

Reply:
Should be resolved by 5bf438a